system/inc/active_object.h

```diff
@@ -280,6 +280,10 @@ class ActiveObjectBase
         return _thread_id == std::this_thread::get_id();
     }
 
+    void setCurrentThread() {
```

Comment:
This has to be public since it's called from main.cpp.

I thought of calling this `attach` or `attachToCurrentThread`. Would that make more sense?